src/anyio/_backends/_asyncio.py

```diff
@@ -385,8 +396,11 @@ def __init__(self, deadline: float = math.inf, shield: bool = False):
         self._cancel_handle: asyncio.Handle | None = None
         self._tasks: set[asyncio.Task] = set()
         self._host_task: asyncio.Task | None = None
-        self._cancel_calls: int = 0
-        self._cancelling: int | None = None
+        self._pending_uncancellations: int | None
+        if sys.version_info >= (3, 11):
+            self._pending_uncancellations = 0
+        else:
+            self._pending_uncancellations = None
```

Comment:
Are you sure we're okay with not looking at the task cancellation counter when entering the scope? I think there was a reason why `asyncio.timeout()` does that.

Reply:
Potentially related? https://github.com/python/cpython/pull/102815/files#diff-0a6da8818df8b29c3c649167aa37c906eb27283c5fe1cacb40e36a6ce0458e7eR87

Reply:
It's *possible* that we don't need this, as we rely on the cancel messages rather than the cancel counter, as I believe you meant by that previous comment of yours.